packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/StocktakeLineEdit.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { FC } from 'react';
+import { StocktakeController, StocktakeItem } from '../../../../types';
+import { ModalMode } from '../../DetailView';
+import { StocktakeLineEditForm } from './StocktakeLineEditForm';
+import {
+  Divider,
+  TableContainer,
+  TabContext,
+  TabList,
+  Tab,
+  alpha,
+  TabPanel,
+  styled,
+  useTranslation,
+  useIsMediumScreen,
+  ButtonWithIcon,
+  PlusCircleIcon,
+  Box,
+} from '@openmsupply-client/common';
+import { BatchTable, PricingTable } from './StocktakeLineEditTables';
+import { createStocktakeRow, wrapStocktakeItem } from './utils';
+
+interface StocktakeLineEditProps {
+  item: StocktakeItem | null;
+  onChangeItem: (item: StocktakeItem | null) => void;
+  mode: ModalMode;
+  draft: StocktakeController;
+}
+
+const StyledTabPanel = styled(TabPanel)({
+  height: '100%',
+});
+
+enum Tabs {
+  Batch = 'Batch',
+  Pricing = 'Pricing',
+}
+
+export const StocktakeLineEdit: FC<StocktakeLineEditProps> = ({
```

Comment:
perhaps when adding an item, automatically add a row on item selection?

Reply:
Generally will be counting the already existing stock lines and adding a new arbitrary batch would be a secondary action, I think. A mechanism to add existing stock lines to count would be preferable? Just not sure how exactly, as you also may not want to count some stock lines 🤔